pkg/apiserver/registry/networkpolicy/clustergroupmember/rest.go

```diff
@@ -70,9 +81,52 @@ func (r *REST) Get(ctx context.Context, name string, options *metav1.GetOptions)
 		memberList.EffectiveMembers = effectiveMembers
 	}
 	memberList.Name = name
-	return memberList, nil
+	memberList.TotalMembers = int64(len(memberList.EffectiveMembers))
+	err = paginateMemberList(memberList, pageInfo)
+	return memberList, err
+}
+
+func (r *REST) NewGetOptions() (runtime.Object, bool, string) {
+	return &controlplane.PaginationGetOptions{}, false, ""
 }
 
 func (r *REST) NamespaceScoped() bool {
 	return false
 }
+
+// paginateMemberList returns paginated results if rational options are provided.
+// Paginated results are continuous only when there is no member change across multiple calls.
+// Returns full memberList if page number or page limit are given incorrectly.
+func paginateMemberList(memberList *controlplane.ClusterGroupMembers, pageInfo paginationInfo) error {
+	if memberList.TotalMembers == 0 {
+		memberList.TotalPages, memberList.CurrentPage = 0, 0
+		return nil
+	}
+	if pageInfo.pageLimit <= 0 {
+		memberList.TotalPages, memberList.CurrentPage = 1, 1
+		return nil
+	}
+	totalPages := (int64(len(memberList.EffectiveMembers)) + pageInfo.pageLimit - 1) / pageInfo.pageLimit
+	if totalPages >= pageInfo.pageNumber && pageInfo.pageNumber > 0 {
+		// Sort members based on name of ee/pod to realize consistent pagination support.
+		sort.SliceStable(memberList.EffectiveMembers, func(i, j int) bool {
+			if memberList.EffectiveMembers[i].Pod != nil && memberList.EffectiveMembers[j].Pod != nil {
+				return memberList.EffectiveMembers[i].Pod.Name < memberList.EffectiveMembers[j].Pod.Name
+			} else if memberList.EffectiveMembers[i].ExternalEntity != nil && memberList.EffectiveMembers[j].ExternalEntity != nil {
+				return memberList.EffectiveMembers[i].ExternalEntity.Name < memberList.EffectiveMembers[j].ExternalEntity.Name
+			} else {
+				return true
+			}
+		})
```

Comment:
Question about behaviour: If the user does not require pagination, the response is not sorted. On the other hand, if the user requires pagination, the response is sorted by Pod name or External entity. Is my understanding correct?

Reply:
Yes that is the case, to reduce extra sorting when user does not require pagination.